platform/testing/end-to-end/cypress-upgrade-guide.md

```diff
@@ -0,0 +1,182 @@
+# Cypress 6.2.1 Migration Guide
+
+The purpose of this guide is to help with migrating existing Cypress tests to apply changes introduced by upgrading from Cypress 4.8 to Cypress 6.2.1.
+
+# Table of Contents
+
+1. [Overview](#overview)
+2. [Migration changes](#migration-changes)
+   1. [`cy.intercept()` vs. `cy.route()`](#intercept)
+   2. [Fixtures](#fixtures)
+   3. [Non-existent elements](#non-existent-elements)
+   4. [Opacity of elements](#opacity)
+   5. [Retrying tests](#retries)
+   6. [Retrying individual tests](#individual-retries)
+   7. [Cookies](#cookies)
+   8. [Blocking hosts](#blocking-hosts)
+   9. [`dirname` and `filename`](#dirname-filename)
+3. [Additional resources](#additional-resources)
+
+## Overview <a name="overview"></a>
+
+The upgrade from Cypress 4.8 to Cypress 6.2.1 introduces several changes to test structure, syntax, and other aspects of Cypress tests. This guide covers key changes that are most likely to affect existing tests. There are a handful of deprecated methods that should be updated before the core Cypress library no longer supports them. A full, detailed guide of all changes can be found in the [additional resources](#additional-resources) section.
+
+## Migration changes <a name="migration-changes"></a>
+
+### `cy.intercept` vs. `cy.route` <a name="intercept"></a>
+
+[`cy.server()`](https://docs.cypress.io/api/commands/server.html) and [`cy.route()`](https://docs.cypress.io/api/commands/route.html) are being deprecated in favor of [`cy.intercept()`](https://docs.cypress.io/api/commands/intercept.html). `cy.server()` is not needed when using `cy.intercept()`. Syntax for `cy.intercept()` is the same as it is for `cy.route()`.
+
+For example, this block of code:
+
+```javascript
+cy.server();
+cy.route('/hello').as('helloWorld');
+```
+
+can be re-written as
+
+```javascript
+cy.intercept('/hello').as('helloWorld');
+```
+
+`cy.wait()` also behaves differently with `cy.intercept()`:
+
+```javascript
+cy.server();
+cy.route('/hello').as('helloWorld');
+cy.wait('@helloWorld')
+  .then(({ requestBody, responseBody, status }) => {
+    expect(status).to.eq(200);
+  });
+```
+
+now looks like:
+
+```javascript
+cy.intercept('/hello').as('helloWorld');
+cy.wait('@helloWorld')
+  .then(({ request, response }) => {
```

Comment:
Did the shape of the object parameter used in the callback change with the version upgrade?

It looks like the shape used to be `{requestBody: ..., responseBody: ..., status: ...}`, and now the shape appears to be `{request: ..., response: { statusCode: ... }}`?

Reply:
It did, the structure seems simpler with the new way.